rasa/utils/tensorflow/rasa_layers.py

```diff
@@ -30,7 +31,165 @@
 from rasa.utils.tensorflow.transformer import TransformerEncoder
 
 
-class ConcatenateSparseDenseFeatures(tf.keras.layers.Layer):
+class RasaCustomLayer(tf.keras.layers.Layer):
+    """Parent class for all classes in `rasa_layers.py`.
+
+    Allows adjusting `DenseForSparse` layers to all the child classes.
+
+    During fine-tuning sparse feature sizes might change due to addition of new data.
+    If this happens, we need to adjust our `DenseForSparse` layers to a new size.
+    `ConcatenateSparseDenseFeatures`, `RasaSequenceLayer` and
+    `RasaFeatureCombiningLayer` all inherit from `RasaCustomLayer` and thus can
+    change their own `DenseForSparse` layers if it's needed.
+    """
+
+    def adjust_sparse_layers_for_incremental_training(
+        self,
+        new_sparse_feature_sizes: Dict[Text, Dict[Text, List[int]]],
+        old_sparse_feature_sizes: Dict[Text, Dict[Text, List[int]]],
+        reg_lambda: float,
+    ) -> None:
+        """Finds and adjusts `DenseForSparse` layers during incremental training.
+
+        Recursively looks through the layers until it finds all the `DenseForSparse`
+        ones and adjusts those which have their sparse feature sizes increased.
+
+        Args:
+            new_sparse_feature_sizes: sizes of current sparse features.
+            old_sparse_feature_sizes: sizes of sparse features the model was
+                                      trained on before.
+            reg_lambda: regularization constant.
+        """
+        for name, layer in self._tf_layers.items():
+            if isinstance(layer, RasaCustomLayer):
+                layer.adjust_sparse_layers_for_incremental_training(
+                    new_sparse_feature_sizes=new_sparse_feature_sizes,
+                    old_sparse_feature_sizes=old_sparse_feature_sizes,
+                    reg_lambda=reg_lambda,
+                )
+            elif isinstance(layer, layers.DenseForSparse):
+                attribute = layer.get_attribute()
+                feature_type = layer.get_feature_type()
+                if (
+                    attribute in new_sparse_feature_sizes
+                    and feature_type in new_sparse_feature_sizes[attribute]
+                ):
+                    new_feature_sizes = new_sparse_feature_sizes[attribute][
+                        feature_type
+                    ]
+                    old_feature_sizes = old_sparse_feature_sizes[attribute][
+                        feature_type
+                    ]
+                    self._check_sparse_feature_decreased_sizes(
+                        new_sparse_feature_sizes=new_feature_sizes,
+                        old_sparse_feature_sizes=old_feature_sizes,
+                    )
+                    if sum(new_feature_sizes) > sum(old_feature_sizes):
+                        new_layer = self._create_dense_for_sparse_layer(
```

Comment:
You can directly do - `self._tf_layers[name] = self._create_dense_for_sparse_layer(....`